projects/redemption_configs/redemption_src/configs/io.hpp

```diff
@@ -424,17 +424,33 @@ parse_from_cfg(T & x, ::configs::spec_type<bool> /*type*/, zstring_view value)
 inline parse_error parse_from_cfg(int& x, ::configs::spec_type<::configs::spec_types::file_permission> /*type*/, zstring_view value)
 {
     int tmp = 0;
+    const char *error_msg = nullptr;
+    bool error_from_user_cfg = false;
 
     if (auto res = std::from_chars(value.begin(), value.end(), tmp, 8);
         res.ptr != value.end() || res.ec != std::errc())
     {
-        return parse_error { "Cannot parse file permission because it's not a octal number" };
+        error_msg = "Cannot parse file permission because it's not an octal number";
+        error_from_user_cfg = true;
     }
     else if (tmp > 0777)
     {
-        return parse_error { "Cannot have file permission higher than 777 octal number" };
+        error_msg = "Cannot have file permission higher than 0777 number";
+        error_from_user_cfg = true;
+    }
+    if (error_from_user_cfg)
+    {
+        if (x > 0777)
```

Comment:
A parser must not check the value of `x`.

Reply:
Christophe told me to check it if file_permission from user is bad